### PR TITLE
Add assignment 0, fix some things

### PR DESCRIPTION
**Pull request by osnr**
- set up local testing (run `sudo gem install bundler` then `bundle install` and then `bundle exec jekyll serve` and it'll serve on localhost:4000) and fix associated things so it works the same as GH pages
- get rid of extra README.md
- move assignments FAQ to /assignment/ folder and also make it an index of released assignments
- turn schedule table into Markdown so it's actually editable
- tweak jekyll config to use kramdown w/ github flavored markdown option so Markdown parser works properly on assign0 writeup (anchors on headings, fenced code blocks)

review? @SergioBenitez @ibush

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -33,6 +33,11 @@
 
   <link rel="stylesheet" href="/noita/app.css">
   <script src="/noita/js/vendor/modernizr.js"></script>
+  <style>
```
Comment (SergioBenitez) [inline _layouts/default.html]: Add stylesheet code to noita/app.scss.

**Comment by phanrahan** [comment]
I like the idea of the data-driven approach. I find both the html and markdown hard to edit as well.

**Comment by osnr** [comment]
I missed the cleanup you did on the HTML earlier today; agree with you that it's better than the Markdown table format, so changed it back.

The data thing sounds good to me, but I don't want to block releasing the assignment on it, so will leave until after if you want to take it on.

It's sort of true that GitHub uses redcarpet, but they use some [custom filters](https://github.com/github/markup/blob/master/README.md) instead of redcarpet's built-in extensions to do stuff like generating heading IDs. I just tried it and redcarpet extensions do the heading->ID conversion differently from GitHub's on-site filters, breaking some of the links in the assignment 0 writeup (headings with periods, apostrophes, etc). So unless there's a specific problem with kramdown, we should switch to it IMO (it does things the same way as GitHub).

Let me know if OK to merge.

File: assignment/assign0/README.md [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    - [Step 6.2: Create a pull request](#step-62-create-a-pull-request)
+- [Afterward: Grader's code review](#afterward-graders-code-review)
+
+#### Step 0: Prerequisites
+
+This assignment expects you to know how to use the UNIX command line
+on Linux or OS X, and how to edit a text file.
+
+If you're on Windows, you should have set up
+[a Linux virtual machine](https://github.com/cs107e/courseware/blob/master/guides/vmware.md);
```
Comment (ibush) [inline assignment/assign0/README.md]: Should link to the website page instead of the markdown

File: assignment/assign0/README.md [hunk 3]
```diff
@@ -0,0 +1,354 @@
...
+you'll be using that machine for all assignments, including this one.
+
+See
+[our UNIX guide](https://github.com/cs107e/courseware/blob/master/guides/unix.md)
```
Comment (ibush) [inline assignment/assign0/README.md]: Fix link

File: assignment/assign0/README.md [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+if you need information about these topics. The standard CS 107 will also
+have a UNIX lab in week 1 which we'll announce to the class, should
+you want to attend.
+
+You should also read through the other
+[course guides](https://github.com/cs107e/courseware/tree/master/guides)
```
Comment (ibush) [inline assignment/assign0/README.md]: Fix link

**Comment by ibush** [comment]
Omar, I think it looks good to go -- just fix the couple guide link issues

**Pull request merged**